File: services/server/src/safe_socket/safe_socket.py
import socket

def recv_all(socket: socket.socket, size):
    buffer = b""
    while len(buffer) < size:
        chunk = socket.recv(size - len(buffer))
        if not chunk:
            if not buffer:
                return b""
            raise RuntimeError("Socket connection closed")
        buffer += chunk
    return buffer


def send_all(socket: socket.socket, bytes):
    sent = 0
    while sent < len(bytes):
        sent += socket.send(bytes[sent:])
    return sent

File: services/client/src/safe_socket/safe_socket.go
package safe_socket

import "io"

//TODO: Complete with a short-read/short-write tolerant implementation

func SendAll(socket io.Writer, bytes []byte) error {
	sent := 0
	for sent < len(bytes) {
		n, err := socket.Write(bytes[sent:]) //controla el caso de q cierre justo
		if err != nil {
			return err
		}
		sent += n
	}
	return nil
}

func RecvAll(socket io.Reader, size int) ([]byte, error) {
	buff := make([]byte, size)
	read := 0
	for read < size {
		n, err := socket.Read(buff[read:])
		read += n
		if err != nil {
			if err == io.EOF && read == size {
				return buff, nil
			}
			return nil, err
		}
	}
	return buff, nil

}
